src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -988,22 +987,29 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
                     arrayItemsCount = row.SizeOrLength;
                     numberOfRowsForValues += row.NumberOfRows;
                 }
-                else if (tokenType == JsonTokenType.PropertyName)
+                else if (IsStringLike(tokenType)) // Equivalent to (tokenType == JsonTokenType.PropertyName || tokenType == JsonTokenType.String)
                 {
                     numberOfRowsForValues++;
                     numberOfRowsForMembers++;
-                    database.Append(tokenType, tokenStart, reader.ValueSpan.Length);
+                    Debug.Assert(reader.TokenStartIndex < int.MaxValue); // Adding 1 to skip the start quote will never overflow
```

Comment:
I can't tell what this trailing comment is modifying.  Please move it to a aligned-with-code comment before the code it's commenting.